src/openrct2/world/MapGen.cpp

```diff
@@ -660,23 +660,25 @@ bool mapgen_load_heightmap(const utf8* path)
     try
     {
         auto image = Imaging::ReadFromFile(path, format);
-        if (image.Width != image.Height)
+
+        auto wSize = image.Width;
+        if (image.Width > MAXIMUM_MAP_SIZE_PRACTICAL)
         {
-            context_show_error(STR_HEIGHT_MAP_ERROR, STR_ERROR_WIDTH_AND_HEIGHT_DO_NOT_MATCH, {});
-            return false;
+            context_show_error(STR_HEIGHT_MAP_ERROR, STR_ERROR_HEIHGT_MAP_TOO_BIG, {});
```

Comment:
This message could be triggered twice now, once for the width and once for the height, causing the audio error to be extra loud. Instead of handling the error here, check if the width and size match later on and only then trigger this message.

```cpp
if (wSize != image.Width || hSize != image.Height)
{
    context_show_error(STR_HEIGHT_MAP_ERROR, STR_ERROR_HEIHGT_MAP_TOO_BIG, {});
}
```